File: src/main/kotlin/demo/kotliln/boardgame/service/boardgame/BoardGameRepository.java
package demo.kotliln.boardgame.service.boardgame;

import demo.kotliln.boardgame.service.entities.BoardGameEntity;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface BoardGameRepository extends CrudRepository<BoardGameEntity, Long> {

    Optional<BoardGameEntity> findByName(String name);

    List<BoardGameEntity> findAll();
}

File: src/main/kotlin/demo/kotliln/boardgame/service/boardgame/repositories/BoardGameRepository.kt
package demo.kotliln.boardgame.service.boardgame.repositories

import demo.kotliln.boardgame.service.boardgame.entities.BoardGameEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface BoardGameRepository : CrudRepository<BoardGameEntity, Long> {
    fun findByName(name: String): Optional<BoardGameEntity>

    override fun findAll(): List<BoardGameEntity>
}